src/platform/src/redox/mod.rs

```diff
@@ -184,6 +184,17 @@ pub fn read(fd: c_int, buf: &mut [u8]) -> ssize_t {
     e(syscall::read(fd as usize, buf)) as ssize_t
 }
 
+pub fn rename(old: *const c_char, new: *const c_char) -> c_int {
+    let (old_path, new_path) = unsafe { (c_str(old), c_str(new)) };
+    match syscall::open(old_path, O_WRONLY) {
+        Ok(fd) => match syscall::frename(fd, new_path) {
```

Comment:
You need to close the fd if frename fails

Reply:
:+1: good catch

Reply:
fixed

Reply:
Also, the return value from frename should be returned, not the one from close

Reply:
It would be better to do:

let res = syscall::frename(fd, new_path);
let _ = syscall::close(fd);
e(res)

Reply:
:+1: updated